File: codechef/FEB19B/p1.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
  long long itr;
  cin>>itr;
  while(itr--){
    long long n,a,b,k;
    cin>>n>>a>>b>>k;
    long long c = n/a + n/b - 2*(n/((a*b)/__gcd(a, b)));
    if(c>=k && a!=b)
      cout<<"Win\n";
    else
      cout<<"Lose\n";
  }
  return 0;
}

File: codechef/FEB19B/p1.py
from math import floor, gcd
itr = int(input())
for i in range(itr):
    l = input().split()
    n,a,b,k = map(int, l)
    cf = (a*b)/gcd(a, b)
    s = int(floor(n/a))+int(floor(n/b))-(2*(int(floor(n/cf))))
    if s>=k and a!=b:
        print("Win")
    else:
        print("No")
